cmd/kyma/apply/function/opts.go

```diff
@@ -0,0 +1,117 @@
+package function
+
+import (
+	"fmt"
+	"github.com/kyma-incubator/hydroform/function/pkg/workspace"
+	"github.com/kyma-project/cli/internal/cli"
+	"os"
+	"path"
+)
+
+//Options defines available options for the command
+type Options struct {
+	*cli.Options
+
+	OnError  value
+	Output   value
+	Filename filename
+	DryRun   bool
+}
+
+//NewOptions creates options with default values
+func NewOptions(o *cli.Options) *Options {
+	return &Options{
+		Options: o,
+		OnError: newValue(NothingOnError, validOnError),
+		Output:  newValue(TextOutput, validOutput),
+	}
+}
+
+var (
+	validOnError = []string{
+		NothingOnError,
+		PurgeOnError,
+	}
+	validOutput = []string{
+		NoneOutput,
+		JSONOutput,
+		YAMLOutput,
+		TextOutput,
+	}
+)
+
+const (
+	NothingOnError = "nothing"
+	PurgeOnError   = "purge"
+)
+
+const (
+	NoneOutput = "none"
+	JSONOutput = "json"
+	YAMLOutput = "yaml"
+	TextOutput = "text"
+)
+
+type value struct {
+	value     string
+	available []string
+}
+
+func newValue(defaultVal string, available []string) value {
+	return value{
+		value:     defaultVal,
+		available: available,
+	}
+}
+
+func (g *value) String() string {
+	return g.value
+}
+
+func (g *value) Set(v string) error {
+	if g == nil {
+		return fmt.Errorf("nil pointer reference")
+	}
+	if v == "" {
+		return nil
+	} else if g.isOneOf(v) {
+		g.value = v
+		return nil
+	}
+	return fmt.Errorf("specified value: %s is not supported", v)
+}
+
+func (g value) isOneOf(item string) bool {
+	for _, elem := range g.available {
+		if elem == item {
+			return true
+		}
+	}
+	return false
+}
+
+func (g value) Type() string {
+	return "value"
+}
+
+type filename string
```

Comment:
What is the benefit of having this type? It looks like a regular string to me.

Reply:
Now nothing. In the previous version, this type had a defaulting constructor and other methods. But I had a problem with keeping $PWD as the default value, because of generated documentation. I will change it ;)